File: FileStorage.DAL/Model/File.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;
using System.Text;

namespace FileStorage.DAL.Model
{
    public partial class File
    {
        public Guid Id { set; get; }

        //public string Path { set; get; }

        public string FileName { set; get; }

        public FileAccessibility FileAccessibility { set; get; }

        public string ShortLink { set; get; }

        public Guid UserId { set; get; }

        public Guid? FolderId { set; get; }

        public Guid CategoryId { set; get; }

        public virtual User User { set; get; }

        public virtual Folder Folder { set; get; }

        public virtual Category Category { set; get; }

        //public string Path 
        //{
        //    get
        //    {
        //        if(Folder != null)
        //            return Folder.FullPath + "\\" + FileName;
        //        return $@"Files\{UserId}\{FileName}";
        //    }
        //}

        public string Path { set; get; }
    }

    public enum FileAccessibility
    {
        Private, Protected, Public
    }
}

File: FileStorage.Web/ClientApp/src/app/model/file.ts
import { Guid } from 'guid-typescript';

// export class FileModel{
//     constructor(
//         public id : string,
//         public path : string,
//         public fileAccessibility : FileAccessibility,
//         public userId : string,
//         public categoryId : string,
//         public file : File,
//         public fileName : string,
//         public shortLink? : string,
//         public folderId? : string,
//         public categoryName? : string,
//         public folderName? : string,
//         public firstName? : string,
//         public lastName? : string
//     ){}
// }

export class FileModel {
    public id: string = Guid.EMPTY;
    public path: string;
    public fileAccessibility: FileAccessibility = FileAccessibility.Private;
    public userId: string;
    public categoryId: string;
    public file: File;
    public fileName: string;
    public shortLink: string;
    public folderId: string;
    public categoryName: string;
    public folderName: string;
    public firstName: string;
    public lastName: string;
}

export enum FileAccessibility {
    Private,
    Protected,
    Public
}